sdk/keyvault/keyvault-admin/src/lro/restore/operation.ts

```diff
@@ -67,176 +37,130 @@ export interface RestorePollOperationState extends PollOperationState<undefined>
-    }
-    if (error && error.message) {
-      state.isCompleted = true;
-      state.error = new Error(error.message);
-    }
-  }
+    if (!state.isStarted) {
+      const serviceOperation = await this.fullRestore({
+        ...this.requestOptions,
+        restoreBlobDetails: {
+          folderToRestore: folderName,
+          sasTokenParameters: {
+            storageResourceUri: blobStorageUri,
+            token: sasToken
+          }
+        }
+      });
 
-  if (!state.jobId) {
-    state.error = new Error(`Missing "jobId" from the full restore operation.`);
-    state.isCompleted = true;
-    return makeRestorePollOperation(state);
-  }
+      const { startTime, jobId, endTime, error, status, statusDetails } = serviceOperation;
 
-  if (!state.isCompleted) {
-    const serviceOperation = await restoreStatus(client, vaultUrl, state.jobId, {
-      requestOptions
-    });
-    const { endTime, status, statusDetails, error } = serviceOperation;
+      if (!startTime) {
+        state.error = new Error(`Missing "startTime" from the full restore operation.`);
+        state.isCompleted = true;
+        return this;
+      }
 
-    state.endTime = endTime;
-    state.status = status;
-    state.statusDetails = statusDetails;
+      state.isStarted = true;
+      state.jobId = jobId;
+      state.endTime = endTime;
+      state.startTime = startTime;
+      state.status = status;
+      state.statusDetails = statusDetails;
 
-    if (endTime) {
-      state.isCompleted = true;
+      if (endTime) {
+        state.isCompleted = true;
+      }
+      if (error && error.message) {
+        state.isCompleted = true;
+        state.error = new Error(error.message);
+      }
     }
-    if (error && error.message) {
+
+    if (!state.jobId) {
+      state.error = new Error(`Missing "jobId" from the full restore operation.`);
       state.isCompleted = true;
-      state.error = new Error(error.message);
+      return this;
     }
-  }
 
-  return makeRestorePollOperation(state);
-}
+    if (!state.isCompleted) {
+      const serviceOperation = await this.restoreStatus(state.jobId, this.requestOptions);
+      const { endTime, status, statusDetails, error } = serviceOperation;
 
-/**
- * @summary Reaches to the service and cancels the key's operation, also updating the key's poll operation
- * @param [options] The optional parameters, which is only an abortSignal from @azure/abort-controller
- */
-async function cancel(this: RestorePollOperation): Promise<RestorePollOperation> {
-  throw new Error("Canceling the deletion of a key is not supported.");
-}
+      state.endTime = endTime;
+      state.status = status;
+      state.statusDetails = statusDetails;
 
-/**
- * @summary Serializes the create key's poll operation
- */
-function toString(this: RestorePollOperation): string {
-  return JSON.stringify({
-    state: this.state
-  });
-}
+      if (endTime) {
+        state.isCompleted = true;
+      }
+      if (error && error.message) {
```

Comment:
what happens if `error` has an empty `message`? Does it never finish? Do we ignore it?

Reply:
True! I'll change this to `if (error)`. If anything, on January I'll be working on testing this better.

Reply:
I think the error is only valid if it has a message. I'll ask the team!

Reply:
I made an issue! https://github.com/Azure/azure-sdk-for-js/issues/12693 I'll follow up before the next Key Vault release.